WordPress.gitignore

```diff
@@ -1,6 +1,10 @@
 # ignore everything in the root except the "wp-content" directory.
+/*
 !wp-content/
 
+# ignore Mac OS file that stores custom attributes for its containing folder
+.DS_Store
```

Comment:
These rules are covered by [`Global/macOS.gitignore`](https://github.com/github/gitignore/blob/master/Global/macOS.gitignore).

Please omit this change as it is not specific to Wordpress.

Reply:
Agreed, also !.gitignore should be added so we can add the .gitignore to the repo :P

Reply:
+1 on both. I think we should have clear separation of concerns on these files—hence I have some serious doubts if, for example, `*.sql` files exclusion should be here in the first place if it's not something that occurs/is generated naturally on a typical WP install. (Want to bring everyone to attention that own project/personal rules, should probably be applied by oneself individually on each project `.gitignore` alongside these general contrib rules, to avoid the risk of possible workflow enforcement onto others.)

Having each file handle the specific files of each library/component/platform, allows for better combination of several of these files, but also for utils like [gitignore.io](http://gitignore.io).

Reply:
".DS_Store" line will be removed from this push as @shiftkey and @NerdsvilleCEO is right